Modules/posixmodule.c

```diff
@@ -9979,6 +9978,77 @@ os_copy_file_range_impl(PyObject *module, int src, int dst, Py_ssize_t count,
 }
 #endif /* HAVE_COPY_FILE_RANGE*/
 
+#ifdef HAVE_SPLICE
+/*[clinic input]
+
+os.splice
+    src: int
+        Source file descriptor.
+    dst: int
+        Destination file descriptor.
+    count: Py_ssize_t
+        Number of bytes to copy.
+    offset_src: object = None
+        Starting offset in src.
+    offset_dst: object = None
+        Starting offset in dst.
+
+Transfer count bytes from one file descriptor to another, where one of
+the file descriptors must refer to a pipe.
+
+If offset_src is None, then src is read from the current position;
+respectively for offset_dst. The offset associated to the file
+descriptor that refers to a pipe must be None.
+[clinic start generated code]*/
+
+static PyObject *
+os_splice_impl(PyObject *module, int src, int dst, Py_ssize_t count,
+               PyObject *offset_src, PyObject *offset_dst)
+/*[clinic end generated code: output=0a97963bda2dd719 input=50737050f1aa6515]*/
+{
+    off_t offset_src_val, offset_dst_val;
+    off_t *p_offset_src = NULL;
+    off_t *p_offset_dst = NULL;
+    Py_ssize_t ret;
+    int async_err = 0;
+    /* The flags argument is provided to allow
```

Comment:
Feel free to ignore this if you haven't implemented this yet but I noticed you added constants for the splice flags like `SPLICE_F_MORE` but you can't actually pass them in here? Is that intentional, should the constants even be exposed if they can't be used?

Reply:
Yeah, I am missing the implementation because I was preparing some tests for the flags but then I realized that is quite challenging to do so so I probably will just push the implementation directly